test/react-native/javascript.test.ts

```diff
@@ -131,4 +137,455 @@ export default App;`;
+  );
+});`;
+
+      const result = checkAndWrapRootComponent(mod);
+
+      expect(result).toBe(SentryWrapResult.Success);
+      expect(generateCode(mod.$ast).code).toBe(expectedOutput);
+    });
+
+    it('wraps a root app anonymous complex function', () => {
+      const mod = parseModule(`import * as Sentry from '@sentry/react-native';
+
+export default () => {
+  const colorScheme = useColorScheme();
+  const [loaded] = useFonts({
+    SpaceMono: require('../assets/fonts/SpaceMono-Regular.ttf'),
+  });
+
+  useEffect(() => {
+    if (loaded) {
+      SplashScreen.hideAsync();
+    }
+  }, [loaded]);
+
+  if (!loaded) {
+    return null;
+  }
+
+  return (
+    <ThemeProvider value={colorScheme === 'dark' ? DarkTheme : DefaultTheme}>
+      <Stack>
+        <Stack.Screen name="(tabs)" options={{ headerShown: false }} />
+        <Stack.Screen name="+not-found" />
+      </Stack>
+      <StatusBar style="auto" />
+    </ThemeProvider>
+  );
+}`);
+
+      const expectedOutput = `import * as Sentry from '@sentry/react-native';
+
+export default Sentry.wrap(() => {
+  const colorScheme = useColorScheme();
+  const [loaded] = useFonts({
+    SpaceMono: require('../assets/fonts/SpaceMono-Regular.ttf'),
+  });
+
+  useEffect(() => {
+    if (loaded) {
+      SplashScreen.hideAsync();
+    }
+  }, [loaded]);
+
+  if (!loaded) {
+    return null;
+  }
+
+  return (
+    <ThemeProvider value={colorScheme === 'dark' ? DarkTheme : DefaultTheme}>
+      <Stack>
+        <Stack.Screen name="(tabs)" options={{ headerShown: false }} />
+        <Stack.Screen name="+not-found" />
+      </Stack>
+      <StatusBar style="auto" />
+    </ThemeProvider>
+  );
+});`;
+
+      const result = checkAndWrapRootComponent(mod);
+
+      expect(result).toBe(SentryWrapResult.Success);
+      expect(generateCode(mod.$ast).code).toBe(expectedOutput);
+    });
+
+    it('wraps a default class export', () => {
+      const mod = parseModule(`import * as Sentry from '@sentry/react-native';
+
+export default class RootLayout extends React.Component {
+  render() {
+    return (
+      <View>
+        Test App
+      </View>
+    );
+  }
+}`);
+
+      const expectedOutput = `import * as Sentry from '@sentry/react-native';
+
+export default Sentry.wrap(class RootLayout extends React.Component {
+  render() {
+    return (
+      <View>
+        Test App
+      </View>
+    );
+  }
+});`;
+
+      const result = checkAndWrapRootComponent(mod);
+
+      expect(result).toBe(SentryWrapResult.Success);
+      expect(generateCode(mod.$ast).code).toBe(expectedOutput);
+    });
+
+    it('does not wrap a root app component if not found', () => {
+      const mod = parseModule(`import * as React from 'react';
+
+import { View } from 'react-native';
+
+const App = () => {
+  return (
+    <View>
+      Test App
+    </View>
+  );
+};
+
+export { App };`);
+
+      const result = checkAndWrapRootComponent(mod);
+
+      expect(result).toBe(SentryWrapResult.NotFound);
```

Comment:
I would also check that the code was not modified.

Reply:
Good idea @krystofwoldrich!
Added the extra checks with 758bf38fd7b2dead074c1209082289de2076ae4e